lib/src/utils/immersive_mode.dart

```diff
@@ -43,7 +42,7 @@ mixin ImmersiveMode<T extends StatefulWidget> on State<T>
   }
 
   Future<void> _setImmersiveMode() async {
-    if (defaultTargetPlatform == TargetPlatform.android) {
+    if (Theme.of(context).platform == TargetPlatform.android) {
```

Comment:
In that context I'd not use the `Theme.of(context).platform` because we're about to make a system call which is only in android. This is not for styling purposes.
Here the docs says is ok to even use `dart:io.Platform` but I guess we can also use `defaultTargetPlatform`.

Reply:
nice catch! will revert in latest commit